src/pygeom.c

```diff
@@ -150,7 +150,8 @@ PyTypeObject GeometryType = {
 /* Get a GEOSGeometry pointer from a GeometryObject, or NULL if the input is
 Py_None. Returns 0 on error, 1 on success. */
 char get_geom(GeometryObject *obj, GEOSGeometry **out) {
-    if (!PyObject_IsInstance((PyObject *) obj, (PyObject *) &GeometryType)) {
+    PyTypeObject *type = ((PyObject *)obj)->ob_type;
+    if ((type != &GeometryType) & (type->tp_base != &GeometryType)) {
```

Comment:
Is this change actually needed? 
Because I experimented with this before (adding the TPFLAGS_BASETYPE), and I seemed to remember the ufuncs worked on the subclasses without any additional change (at least in python, "isinstance" works for subclasses as well, but don't know if the semantics of the C API version is exactly the same).

Reply:
`PyObject_IsInstance` [needs the GIL](https://github.com/python/cpython/blob/efc28bbbeee07867669dfbf49385d6a2147fe631/Objects/abstract.c#L2535) while the new typecheck doesn't (it operates on structs)

Looking further into the source, it seems that the "exact type match" (which we always have in current master) is catched very soon in the `PyObject_IsInstance` ([here](https://github.com/python/cpython/blob/efc28bbbeee07867669dfbf49385d6a2147fe631/Objects/abstract.c#L2475) and [here](https://github.com/python/cpython/blob/d905df766c367c350f20c46ccd99d4da19ed57d8/Include/object.h#L127)) so that's why we probably got away with it in the first place.